Add progress bars and FIt-sne

saketkc opened the pull request:
This PR does the following

- Replaces print with progress bars (Demo: https://gist.github.com/saketkc/708d8f03b8d42090fd3dd4a8ff8feba5)
- Adds FIt-sne as a layout option (https://github.com/KlugerLab/pyFIt-SNE)

duhaime commented:
@saketkc Thanks very much for this PR! I love the progress bars, and very much like the idea of the fitsne.

I wanted to ask, though, how do you feel about making the `fitsne` library an optional dependency? I ask because it has a dependency on header files that are not shipped with the library itself. We could create `utils/fitsne-dependencies.txt` and list `fitsne` there, so those who want to kick the tires have an easier time getting started. @pleonard212 and I just discussed the notion of creating a similar `utils/gpu-dependencies.txt` for those who want to run the processing with GPU-acceleration. We'd just want to update the README to tell users how to use the fitsne layout, and wrap the import statements in `utils/process_images.py` in a try/catch block.

I also wanted to ask, what is `config`? You pass it to the tf.Session constructor but it doesn't seem to be defined.

Last, what do you think about adding progress bars for the print statements you commented out? I think those could be nice to have...

saketkc commented:
Thanks @duhaime! I have tried to address your comments:
- Made fitsne optional
- Added `utils/fitsne-dependencies.txt`
- Cleaned up `config` which was unused from `tf.Session`

duhaime commented:
Sorry for the insane delay on this @saketkc. I'm just circling back now and this is fantastic! Thanks very much!
